R/jira.R

```diff
@@ -765,3 +765,449 @@ create_status <- function(jira_domain_url, status) {
+#' will then be downloaded. This allows us to download only issues that have not already been
+#' downloaded previously. If the directory is empty, then it will download all issues.
+#'
+#' @param domain Custom JIRA domain URL set in config file
+#' @param credentials a path to text file containing your username/api token
+#' @param jql_query Specific query string to specify criteria for fetching
+#' @param fields List of fields that are downloaded in each issue
+#' @param save_folder_path Path that files will be save along in .json format. Includes the prefix for projectname_issues
+#' @param max_results (optional) the maximum number of results to download per page.
+#' Default is 50. [How to use max_results](https://confluence.atlassian.com/jirakb/how-to-use-the-maxresults-api-parameter-for-jira-issue-search-rest-api-1251999998.html)
+#' @param max_total_downloads Maximum downloads per function call.
+#' This value specifies how many issues should be extracted and API calls will cease if a subsequent
+#' API call would reach or surpass this value.
+#' @param unaltered_file_path the path to the directory that contains the files
+#' @param verbose boolean flag to specify printing operational
+#' messages or not. These messages will describe errors or successes in executing code.
+#' And example may be declaring the file name of a successfully downloaded file or printing
+#' how many issues have been downloaded so far
+#' @export
+#' @family downloaders
+#' @family jira
+#' @seealso  \code{link{download_jira_issues_by_issue_key}} to download JIRA data using issuekey as a search criteria,
+#' @seealso  \code{link{download_jira_issues}} to download all JIRA issues data,
+#' @seealso  \code{link{download_jira_issues_by_date}} to download JIRA data using created as a search criteria,
+#' @seealso  \code{link{parse_jira}} to parse jira issues along a file-path,
+#' @seealso \code{link{parse_jira_latest_date}} to retrieve the file_name to be passed to this function
+refresh_jira_issues <- function(domain,
+                                credentials,
+                                jql_query,
+                                fields,
+                                save_folder_path,
+                                max_results,
+                                max_total_downloads,
+                                unaltered_file_path,
+                                verbose){
+
+  # Check if the file exists
+  if(file.exists(unaltered_file_path)) {
+    # Check if the file is empty by checking its size
+    # List all files and subdirectories in the directory
+    contents <- list.files(path = unaltered_file_path)
+
+    # If the file is empty, download all issues
+    if(length(contents) == 0) {
+      if(verbose){
+        message("The file is empty.Downloading all\n")
+      }
+      download_jira_issues(domain,
+                           credentials,
+                           jql_query,
+                           fields,
```

Comment:
The `fields` parameter should include a default containing the fields the parser is expected to handle. Will check on this in a bit.